lib/ansible/modules/network/onyx/onyx_ntp_servers_peers.py

```diff
@@ -0,0 +1,275 @@
+                         delete=dict(choices=['yes', 'no']))
+        server_spec = dict(ip_or_name=dict(required=True),
+                           state_enabled=dict(choices=['yes', 'no']),
+                           version=dict(type='int', choices=[3, 4]),
+                           trusted_enable=dict(choices=['yes', 'no']),
+                           key_id=dict(type='int'),
+                           delete=dict(choices=['yes', 'no']))
+        element_spec = dict(peers=dict(type='list', elements='dict', options=peer_spec),
+                            servers=dict(type='list', elements='dict', options=server_spec),
+                            ntpdate=dict())
+        argument_spec = dict()
+        argument_spec.update(element_spec)
+        self._module = AnsibleModule(
+            argument_spec=argument_spec,
+            supports_check_mode=True)
+
+    def get_required_config(self):
+        module_params = self._module.params
+        self._required_config = dict(module_params)
+        self.validate_param_values(self._required_config)
+
+    def _show_peers_servers_config(self):
+        cmd = "show ntp configured"
+        return show_cmd(self._module, cmd, json_fmt=True, fail_on_error=False)
+
+    def _set_servers_config(self, peers_servers_config):
+        servers = dict()
+        peers = dict()
+        if not peers_servers_config:
+            return
+        index = 0
+        for peer_server in peers_servers_config:
+            if (index == 0):
+                index += 1
+                continue
+            else:
+                header_list = peer_server.get("header").split(" ")
+                type = header_list[1]
+                if (type == 'server'):
+                    server_entry = {"version": peer_server.get("NTP version"),
+                                    "state_enabled": peer_server.get("Enabled"),
+                                    "trusted_enable": peer_server.get("Trusted"),
+                                    "key_id": peer_server.get("Key ID")}
+                    servers.update({header_list[2]: server_entry})
+                else:
+                    peer_entry = {"version": peer_server.get("NTP version"),
+                                  "state_enabled": peer_server.get("Enabled"),
+                                  "key_id": peer_server.get("Key ID")}
+                    peers.update({header_list[2]: peer_entry})
+            index += 1
+            self._current_config.update({"servers": servers})
+            self._current_config.update({"peers": peers})
+
+    def load_current_config(self):
+        self._current_config = dict()
+        peers_servers_config = self._show_peers_servers_config()
+        if peers_servers_config:
+            self._set_servers_config(peers_servers_config)
+
+    def generate_commands(self):
+        ntp_options = ["peers", "servers"]
+        for option in ntp_options:
+            option_name = "peer"
+            if option == "servers":
```

Comment:
no need for this mapping, you can change the current config dictionary to be aligned with the required config or vise-versa